devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -251,6 +251,15 @@ public void execute() throws MojoFailureException, MojoExecutionException {
                 } else {
                     localProject = LocalProject.loadWorkspace(outputDirectory.toPath());
                     for (LocalProject project : localProject.getSelfWithLocalDeps()) {
+                        if (project.getClassesDir() != null) {
+                            //if this project also contains Quarkus extensions we do no want to include these in the discovery
+                            //a bit of an edge case, but if you try and include a sample project with your extension you will
+                            //run into problems without this
+                            if (Files.exists(project.getClassesDir().resolve("META-INF/quarkus-extension.properties")) ||
```

Comment:
Should we add these into a list of something? 
Probably not useful now, but we may see more exceptions like these. 

WDYT?

Reply:
That's being said I believe it's fine for now.

Reply:
If we get any more it should be a list. This is very minor, and did not actually help with Martin's case, but it might help for others.